custom_components/plugwise/number.py

```diff
@@ -14,32 +15,31 @@
 from homeassistant.const import TEMP_CELSIUS, EntityCategory
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from plugwise import Smile
+from plugwise import DeviceData, Smile
 
 from .const import COORDINATOR  # pw-beta
 from .const import DOMAIN, LOGGER
 from .coordinator import PlugwiseDataUpdateCoordinator
 from .entity import PlugwiseEntity
 
+T = TypeVar("T", bound=DeviceData)
+
 
 @dataclass
-class PlugwiseEntityDescriptionMixin:
+class PlugwiseNumberMixin:
     """Mixin values for Plugwse entities."""
 
     command: Callable[[Smile, str, float], Awaitable[None]]
+    max_value_fn: Callable[[T], float]
+    min_value_fn: Callable[[T], float]
+    step_key_fn: Callable[[T], float]
+    value_fn: Callable[[T], float]
 
 
 @dataclass
-class PlugwiseNumberEntityDescription(
-    NumberEntityDescription, PlugwiseEntityDescriptionMixin
-):
+class PlugwiseNumberEntityDescription(NumberEntityDescription, PlugwiseNumberMixin):
     """Class describing Plugwise Number entities."""
 
-    native_max_value_key: str | None = None
-    native_min_value_key: str | None = None
-    native_step_key: str | None = None
-    native_value_key: str | None = None
-
```

Comment:
Please change back using native in the varnames as indicated in comment on Discord.

Reply:
👍